### PR TITLE
Added Lua

### LINKED ISSUE #69: Lua
Progle says

image

### PR DESCRIPTION
**Pull request by ricardoboss**
Fixes #69

### CODE REVIEW

**Comment by chucker** [comment]
>Applications.Scripts

Do OSes ship with a Lua interpreter?

File: Prolangle/Languages/Lua.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+using Prolangle.Languages.Framework;
+
+namespace Prolangle.Languages;
+
+public class Lua : BaseLanguage
+{
+	public override Guid Id { get; } = Guid.NewGuid();
+	public override string Name => "Lua";
+	public override TypeSystem Typing => TypeSystem.Weak | TypeSystem.Dynamic | TypeSystem.Duck;
+	public override bool Compiled => false;
```
Comment (chucker) [inline Prolangle/Languages/Lua.cs:10]: If we do do #77, maybe this one should be JIT.

**Comment by chucker** [comment]
Mostly LGTM but I'm unsure if we should do `Scripts` unless a widespread OS ships Lua. Otherwise, `Scripts` is just a synonym for `Interpreted`…

File: Prolangle/Languages/Lua.cs (hunk 1 above)
Comment (ricardoboss) [reply Prolangle/Languages/Lua.cs:10]: So, postpone this PR until #77 is fixed?

**Comment by ricardoboss** [comment]
> > Applications.Scripts
> 
> Do OSes ship with a Lua interpreter?

It comes preinstalled on macOS and some linux distros...

File: Prolangle/Languages/Lua.cs (hunk 1 above)
Comment (chucker) [reply Prolangle/Languages/Lua.cs:10]: Nah… we can merge this one regardless.

**Comment by chucker** [comment]
> > > Applications.Scripts
> > 
> > 
> > Do OSes ship with a Lua interpreter?
> 
> ~It comes preinstalled on macOS and some linux distros...~
> 
> It is installed on my macOS since I use tools that use Lua.

Huh. Seems I have it because of `nmap`.

Anyway, I guess that's OK.

**Review by chucker — APPROVED** [review]

**Pull request merged**